src/IECoreDelight/ShaderNetworkAlgo.cpp

```diff
@@ -293,6 +293,180 @@ void addDefaultUVShader( ShaderNetwork *shaderNetwork )
 	}
 }
 
+/// \todo This is almost identical (maybe should be completely identical)
+/// to `IECoreArnold::ShaderNetworkAlgo::parameterValue`. Should that get
+/// pulled out to a common location?
+template<typename T>
+T parameterValue( const Shader *shader, InternedString parameterName, const T &defaultValue )
```

Comment:
Ah! This is what I think we need in Renderer.cpp as well. Maybe the easiest thing is to do your `\todo` about moving the geometry handling out of DelightLight, with two new functions in ShaderNetworkAlgo?

```
createLightGeometry( const ShaderNetwork * );
updateLightGeomtry( const ShaderNetwork * );
```

Then all the logic pertaining to USD light conversion is in one place.

Reply:
I added a `createUSDLightGeometry()` and `updateUSDLightGeometry()` as well as `usdLightType()` in the latest push (https://github.com/GafferHQ/gaffer/pull/5603/commits/113439bbfe16014347af11e81baa9aca0ea0b1e9).